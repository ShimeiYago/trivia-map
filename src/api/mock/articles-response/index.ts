import { GetArticlesPreviewsResponse } from 'api/articles-api/get-articles-previews';
import { GetMyArticlesResponse } from 'api/articles-api/get-my-articles';
import { GetArticleResponse } from 'api/articles-api/get-remote-article';
import { PostArticleResponse } from 'api/articles-api/post-remote-article';

export const mockGetArticleResponse: GetArticleResponse = {
  postId: 1,
  title: 'ノーチラス号',
  description: 'ネモ船長が最大の発明だと自負する潜水艇。',
  marker: {
    markerId: 1,
    lat: -111.33,
    lng: 149.243,
    park: 'S',
    numberOfPublicArticles: { total: 4, eachCategory: [1, 2, 1] },
    areaNames: ['シー', 'メディテレーニアンハーバー', 'ポルトパラディーゾ'],
  },
  image: 'https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg',
  isDraft: false,
  author: {
    userId: 1,
    nickname: 'Axel',
    icon: 'https://disneyparkstory.com/wp-content/uploads/2017/09/apple-touch-icon.png',
  },
  createdAt: '2022/06/16 21:42',
  updatedAt: '2022/06/16 21:42',
  category: 1,
  numberOfLikes: 1,
};

export const mockPostArticleResponse: PostArticleResponse = {
  postId: 1,
  title: 'ノーチラス号',
  description: 'ネモ船長が最大の発明だと自負する潜水艇。',
  marker: 100,
  image: null,
  isDraft: false,
  author: 0,
  category: 1,
  createdAt: '2022/06/16 21:42',
  updatedAt: '2022/06/16 21:42',
};

export const mockGetArticlesPreviewsResponse: GetArticlesPreviewsResponse = {
  nextUrl: null,
  previousUrl: null,
  totalRecords: 2,
  totalPages: 2,
  currentPage: 1,
  startIndex: 1,
  endIndex: 1,
  results: [
    {
      postId: 1,
      title: 'ノーチラス号のエンジン',
      image: 'https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg',
      category: 1,
      createdAt: '2022/06/16 21:42',
      numberOfLikes: 1,
    },
    {
      postId: 2,
      title: 'ノーチラス号の名前の由来',
      image: null,
      category: 1,
      createdAt: '2022/06/16 21:42',
      numberOfLikes: 1,
    },
  ],
};

export const mockGetMyArticlesResponse: GetMyArticlesResponse = {
  nextUrl: null,
  previousUrl: null,
  totalRecords: 2,
  totalPages: 2,
  startIndex: 1,
  endIndex: 1,
  currentPage: 1,
  results: [
    {
      postId: 1,
      title: 'ノーチラス号のエンジン',
      isDraft: false,
      category: 1,
    },
    {
      postId: 2,
      title: 'ノーチラス号の名前の由来',
      isDraft: true,
      category: 1,
    },
  ],
};
