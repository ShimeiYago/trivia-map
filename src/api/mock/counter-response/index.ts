import { GetCounterResponse, PostCounterResponse } from '../../counter-api';

export const getCounterMockResponse: GetCounterResponse = {
  count: 5,
};

export const postCounterMockResponse: PostCounterResponse = {
  count: 5,
};
