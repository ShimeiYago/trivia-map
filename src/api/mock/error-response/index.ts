import { AxiosError } from 'axios';

export const axiosMockError = {
  response: {
    status: 500,
    data: {},
  },
  message: 'Intentional API Error with mock',
} as AxiosError;
