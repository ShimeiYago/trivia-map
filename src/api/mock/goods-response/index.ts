import { ToggleGoodResponse } from '../../goods-api/toggle-good';
import { CheckGoodStatusResponse } from 'api/goods-api/check-good-status';

export const mockCheckGoodStatusResponse: CheckGoodStatusResponse = {
  haveAddedGood: true,
};

export const mockToggleGoodResponse: ToggleGoodResponse = {
  haveAddedGood: true,
  numberOfGoods: 2,
};
