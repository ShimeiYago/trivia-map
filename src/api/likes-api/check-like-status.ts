import { API_TIMEOUT, BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockCheckLikeStatusResponse } from '../mock/likes-response';

export async function checkLikeStatus(postId: number): Promise<CheckLikeStatusResponse> {
  const axiosInstance = getAxiosInstance(
    { timeout: API_TIMEOUT.short },
    mockCheckLikeStatusResponse,
  );

  const url = `${BASE_URL}/likes/check/${postId}`;

  try {
    const res: AxiosResponse<CheckLikeStatusResponse> = await axiosInstance.get(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type CheckLikeStatusResponse = {
  haveLiked: boolean;
};
