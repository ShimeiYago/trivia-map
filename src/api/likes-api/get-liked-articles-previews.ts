import { API_TIMEOUT } from '../../constant/index';
import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { getUrlParameters } from 'utils/get-url-parameters';
import { GetArticlesPreviewsResponseEachItem } from 'api/articles-api/get-articles-previews';
import { PaginationResponse } from 'api/types/pagination-response';
import { mockGetLikedArticlesPreviewsResponse } from 'api/mock/likes-response';

export async function getLikedArticlesPreviews(
  param: GetLikedArticlesPreviewsParam,
): Promise<GetLikedArticlesPreviewsResponse> {
  const axiosInstance = getAxiosInstance(
    { timeout: API_TIMEOUT.short },
    mockGetLikedArticlesPreviewsResponse,
  );

  const urlParams = getUrlParameters(param);
  const url = `${BASE_URL}/likes/mine${urlParams}`;

  try {
    const res: AxiosResponse<GetLikedArticlesPreviewsResponse> = await axiosInstance.get(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type GetLikedArticlesPreviewsResponseEachItem = {
  article: GetArticlesPreviewsResponseEachItem;
};

export type GetLikedArticlesPreviewsResponse =
  PaginationResponse<GetLikedArticlesPreviewsResponseEachItem>;

export type GetLikedArticlesPreviewsParam = {
  page?: number;
};
