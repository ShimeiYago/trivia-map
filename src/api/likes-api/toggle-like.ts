import { API_TIMEOUT, BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockToggleLikeResponse } from '../mock/likes-response';

export async function toggleLike(postId: number): Promise<ToggleLikeResponse> {
  const axiosInstance = getAxiosInstance({ timeout: API_TIMEOUT.long }, mockToggleLikeResponse);

  const url = `${BASE_URL}/likes/toggle/${postId}`;

  try {
    const res: AxiosResponse<ToggleLikeResponse> = await axiosInstance.post(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type ToggleLikeResponse = {
  haveLiked: boolean;
};
