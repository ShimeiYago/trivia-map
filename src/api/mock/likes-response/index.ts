import { ToggleLikeResponse } from './../../likes-api/toggle-like';
import { CheckLikeStatusResponse } from 'api/likes-api/check-like-status';
import { GetLikedArticlesPreviewsResponse } from 'api/likes-api/get-liked-articles-previews';

export const mockCheckLikeStatusResponse: CheckLikeStatusResponse = {
  haveLiked: true,
};

export const mockToggleLikeResponse: ToggleLikeResponse = {
  haveLiked: true,
};

export const mockGetLikedArticlesPreviewsResponse: GetLikedArticlesPreviewsResponse = {
  nextUrl: null,
  previousUrl: null,
  totalRecords: 2,
  totalPages: 2,
  currentPage: 1,
  startIndex: 1,
  endIndex: 1,
  results: [
    {
      article: {
        postId: 1,
        title: 'ノーチラス号のエンジン',
        image: 'https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg',
        category: 1,
        createdAt: '2022/06/16 21:42',
        numberOfGoods: 1,
      },
    },
    {
      article: {
        postId: 2,
        title: 'ノーチラス号の名前の由来',
        image: null,
        category: 1,
        createdAt: '2022/06/16 21:42',
        numberOfGoods: 1,
      },
    },
  ],
};
