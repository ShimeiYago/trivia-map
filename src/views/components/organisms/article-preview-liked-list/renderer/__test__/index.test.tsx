import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, Props, State } from '..';
import * as GetLikedArticlesPreviewsModule from 'api/likes-api/get-liked-articles-previews';
import { mockGetLikedArticlesPreviewsResponse } from 'api/mock/likes-response';
import { Location } from 'react-router-dom';

let wrapper: ShallowWrapper<Props, State, Renderer>;
let getArticlesPreviewsSpy: jest.SpyInstance;

const basicProps: Props = {
  searchConditions: {
    marker: 1,
  },
  variant: 'large',
  throwError: jest.fn(),
  refreshUser: jest.fn(),
  page: 0,
  location: { pathname: '/pathname', search: '?category=1' } as Location,
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    wrapper = shallow(<Renderer {...basicProps} />);
    getArticlesPreviewsSpy = jest.spyOn(GetLikedArticlesPreviewsModule, 'getLikedArticlesPreviews');
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('fetchArticlesPreviews', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    getArticlesPreviewsSpy = jest.spyOn(GetLikedArticlesPreviewsModule, 'getLikedArticlesPreviews');
  });

  it('should set success states if api calling succeed', async () => {
    getArticlesPreviewsSpy.mockResolvedValue(mockGetLikedArticlesPreviewsResponse);

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();
    await instance['fetchArticlesPreviews']();

    expect(instance.state.loadingState).toBe('success');
  });

  it('should call throwError if api calling fail', async () => {
    getArticlesPreviewsSpy.mockRejectedValue(new Error());

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();
    await instance['fetchArticlesPreviews']();

    expect(instance.props.throwError).toBeCalled();
  });
});
