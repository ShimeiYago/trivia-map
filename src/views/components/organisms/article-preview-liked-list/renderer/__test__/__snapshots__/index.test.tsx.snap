// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<Fragment>
  <div />
  <ForwardRef(Stack)
    spacing={3}
  >
    <CenterSpinner />
  </ForwardRef(Stack)>
</Fragment>
`;
